### ISSUE TITLE
[image_picker] Resizing + compressing multiple images makes app freeze

### ISSUE THREAD

**Issue by Zazo032** [flutter/flutter#123531]
## Steps to Reproduce

1. Run the provided code with the `image_picker` plugin
2. Select 20+ images taken with the physical device's camera (usually 4000x3000 in size)
3. Confirm selection to go back to the app

**Expected results:** App doesn't freeze (ANR).
**Actual results:** App freezes after confirming the selection from the gallery, app shows a black screen for a few seconds (probably while image is being resized/compressed).

<details>
<summary>Code sample</summary>

```dart
final  images = await ImagePicker().pickMultiImage(
  imageQuality: 90,
  maxHeight: 1920,
  maxWidth: 1920,
  requestFullMetadata: false,
);
```

</details>

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel stable, 3.7.8, on Microsoft Windows [Versi¢n 10.0.22621.1413], locale es-ES)
    • Flutter version 3.7.8 on channel stable at C:\src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 90c64ed42b (6 days ago), 2023-03-21 11:27:08 -0500
    • Engine revision 9aa7816315
    • Dart version 2.19.5
    • DevTools version 2.20.1

[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc2)
    • Android SDK at C:\Users\crist\AppData\Local\Android\sdk
    • Platform android-33, build-tools 34.0.0-rc2
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe


[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] Connected device (4 available)
    • SM A217F (mobile) • RF8R30GZ8DJ • android-arm64  • Android 12 (API 31)
    • Chrome (web)      • chrome      • web-javascript • Google Chrome 111.0.5563.111
    • Edge (web)        • edge        • web-javascript • Microsoft Edge 111.0.1661.54

[√] HTTP Host Availability
    • All required HTTP hosts are available

```

</details>

**Comment by huycozy** [comment]
Hi @Zazo032 
I see your comment at the PR. So, does it mean the linked PR will fix this issue? Also, does the crash stack trace similar to linked issues on that PR (https://github.com/flutter/flutter/issues/94210, https://github.com/flutter/flutter/issues/114080)?

**Comment by Zazo032** [comment]
It improves a bit the performance, but still selecting many images will make the app go into a ANR (Application Not Responding) state for a couple seconds. I think all the I/O operations are currently being done in the main thread, which blocks the UI until all the I/O operations are completed in all of the images. An easy way to try to reproduce this is, try to change `imageQuality`, `maxHeight` and `maxHeight`, and try to select ~20 images. After tapping "Done" in the phone's gallery, the gallery view will stay frozen for ~2s, then it's popped and you see a black screen for other ~4s

**Comment by Zazo032** [comment]
@huycozy just tried the latest changes on the PR - it now completely fixes this issue, so would be great if it can be reviewed and merged soon 👍

**Comment by huycozy** [comment]
Thanks for confirming this. This issue may be a reproducible case of linked issues (which happen on production and can not reproduce locally) on that PR. 

Keeping this issue open until the PR above is merged. Please follow up on it for further updates. Once it's merged but the issue still occurs on new package version, please write in comments and then I will get back and check this. 

Also, it would be appreciated if you can add more information about this issue (crash log, affected `image_picker ` version, demo video, etc ... if any). Thanks!

**Comment by Zazo032** [comment]
There are no crash logs, since the "error" is that the app just freezes and then a "Application Not Responding (ANR)" dialog appears in the device. We were using `image_picker: 0.8.7+1`, but we're now using the fork of the PR and seems to be working fine.

**Comment by huycozy** [comment]
I also can reproduce this issue with the same plugin version when selecting more than 20 large-size images. Verified on `Samsung Galaxy Tab A7 Lite, Android 13 OneUI 5.0, SM-T225`. 

<details>
<summary>ANR screenshot</summary>

<img width="400" src="https://user-images.githubusercontent.com/104349824/228802309-6bef5aee-86d7-4c22-94aa-2da738100972.png">

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.7.8, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.7.8 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 90c64ed42b (34 hours ago), 2023-03-21 11:27:08 -0500
    • Engine revision 9aa7816315
    • Dart version 2.19.5
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.76.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.60.0

[✓] Connected device (3 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64  • Android 11 (API 30)
    • macOS (desktop)  • macos            • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)     • chrome           • web-javascript • Google Chrome 111.0.5563.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.9.0-18.0.pre.51, on macOS 13.0.1 22A400 darwin-x64, locale en-VN)
    • Flutter version 3.9.0-18.0.pre.51 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 659ba386f6 (3 hours ago), 2023-03-28 19:13:50 -0500
    • Engine revision 78f9c68971
    • Dart version 3.0.0 (build 3.0.0-376.0.dev)
    • DevTools version 2.22.2
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.76.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.60.0

[✓] Connected device (3 available)
    • iPhone 8 Plus (mobile) • A6BF67EF-D048-430D-A097-324F146781DA • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • macOS 13.0.1 22A400 darwin-x64
    • Chrome (web)           • chrome                               • web-javascript • Google Chrome 111.0.5563.110

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by reidbaker** [comment]
@tarrinneal do you have thoughts on the priority of this feature. I think you recent work in image_picker has you regularly selecting multiple images. 
Could this issue impact the new permissions feature you are working on?

**Comment by tarrinneal** [comment]
> @tarrinneal do you have thoughts on the priority of this feature. I think you recent work in image_picker has you regularly selecting multiple images. Could this issue impact the new permissions feature you are working on?

That feature has been merged already. They aren't related and shouldn't be a concern.

Seems there's a fix already here https://github.com/flutter/packages/pull/3506

**Comment by Zazo032** [comment]
@huycozy this is fixed with `image_picker_android: ^0.8.6+10`, you might also want to check and close related issues 🙂
